Client/Internal/Request.cs

```diff
@@ -107,12 +106,42 @@ public IRequest WithArgument(string key, object value)
         }
 
         /// <summary>Add HTTP query string arguments.</summary>
-        /// <param name="arguments">The key=>value pairs in the query string. If this is a dictionary, the keys and values are used. Otherwise, the property names and values are used.</param>
+        /// <param name="arguments">An enumeration of key=>value pairs.</param>
+        /// <returns>Returns the request builder for chaining.</returns>
+        /// <example><code>client.WithArguments(new[] { new KeyValuePair&lt;string, string&gt;("genre", "drama"), new KeyValuePair&lt;string, int&gt;("genre", "comedy") })</code></example>
+        public IRequest WithArguments<TKey, TValue>(IEnumerable<KeyValuePair<TKey, TValue>> arguments)
+        {
+            var args = Enumerable.Empty<KeyValuePair<string, object>>().ToArray();
```

Comment:
I suggest not initialising `args` in cases where the value will be discarded, and creating an empty array directly instead of using LINQ. For example:
```cs
KeyValuePair<string, object>[] args;
if (arguments != null)
{
    args = arguments.GetType()
        .GetRuntimeProperties()
        .Where(p => p.CanRead)
        .Select(p => new KeyValuePair<string, object>(p.Name, p.GetValue(arguments)))
        .ToArray();
}
else
    args = new KeyValuePair<string, object>[0];
```

Not specific to your code, but we can also ignore indexers to avoid `TargetParameterCountException` in some cases:
```cs
        .Where(p => p.CanRead && p.GetIndexParameters() == null)
```